left sidebar : Add 'Display starred count' option in starred messages popover

ruchit2801 opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
This is followup PR of #11645. Adds option to toggle displaying counts for starred messages in popover of Starred messages

**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
![ScreenRecording20190322at1](https://user-images.githubusercontent.com/39343253/54807099-26bf6080-4ca2-11e9-9835-d80bcbb572db.gif)

ruchit2801 commented:
@showell @rishig Any suggestion for a good icon here?

rishig commented on static/templates/starred_messages_sidebar_actions.handlebars:
```diff
@@ -6,4 +6,14 @@
+        <div id="toggle-starred-counts-status"></div>
+        <a id="toggle_display_starred_msg_count">
+            {{#if starred_message_counts}}
+                {{#tr this}}Hide starred message counts{{/tr}}
```
LEt's change counts -> count on both of tehse

rishig commented:
For icon we can use font awesome's eye-with-a-line icon (i.e. the thing we use for mute topic in the recipient bar)

ruchit2801 commented:
@rishig Great, added icon and changed to count!

ruchit2801 commented:
I think eye will look more better. I'm also toggling the icon. Just eye when show option and eye with slash when hide option.

ruchit2801 commented:
@timabbott Anything more that needs to be done in this or it's finished?

timabbott commented:
From looking at your screencast, I think there's something wrong with the sizing and/or interaction -- you shouldn't see a briefly line-wrapped state after toggling the option.  I think probably just closing the popover before changing the text would be correct.

ruchit2801 commented:
@timabbott That was probably because of the div that I added before anchor tag for the status. I've changed that element to `left-sidebar-modal-holder`. Now I think it's closing properly. See new screencast I've uploaded

timabbott commented:
@ruchit2801 makes sense; can you put that detail about the success element in a comment?  

Also copying @shubhamdhama who may have feedback on whether it's a bad idea to call `do_settings_change` from here (I would have expected we'd call something lower-level if we don't want the UI feedback)

ruchit2801 commented:
@timabbott I added comment for now. I'll try to find a way to change setting without UI feedback. @shubhamdhama a small hint would be awesome.

shubhamdhama commented:
(Hi Ruchit, I can do a detailed review after tomorrow, I have got an exam
tomorrow)

On Wed, Apr 10, 2019 at 10:52 AM ruchit2801 <notifications@github.com>
wrote:

> @timabbott <https://github.com/timabbott> I added comment for now. I'll
> try to find a way to change setting without UI feedback. @shubhamdhama
> <https://github.com/shubhamdhama> a small hint would be awesome.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zulip/zulip/pull/11940#issuecomment-481537836>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AVNVCFYBj_mRQnHpWooZrCvq1A5dS9NJks5vfXUEgaJpZM4b7hQj>
> .
>

shubhamdhama commented on static/js/stream_popover.js:
```diff
@@ -327,6 +328,19 @@ exports.register_stream_handlers = function () {
+
+        // do_settings_change expects one jQuery element to display success status. We passed
+        // '.left-sidebar-modal-holfer' here. We can create a new one, but that makes no sense
+        settings_ui.do_settings_change(channel.patch, '/json/settings/display', data, '.left-sidebar-modal-holder');
```
Thanks for working on this :)

> In this case we don't want to display any success message on successful toggling.

If this is the case, then yeah definitely, it doesn't make sense to me as well to use `do_settings_change` function, it should only be used when you expect to show that nice "Saving.."/"Saved" mechanism feedback. You should directly use `channel.patch` (you can grep for that) and just pass `url` and `data`.

ruchit2801 replied:
Thanks! "Changing data on backend will send event to update frontend" was the part I missed here :). Earlier what I was afraid of was, not doing setting change will not update in settings section. Updated corresponding line, and now it just uses `channel.patch`. Thanks again :)

ruchit2801 commented:
@timabbott changes pushed, ready to be merged :)

ruchit2801 commented:
@timabbott, I think this can be merged now.

timabbott commented:
Looks great, merged after fixing the commit message to not have a space before the colon, thanks @ruchit2801!  (I think `tools/lint` will complain about that commit message for you; I highly recommend setting up our pre-commit hook to catch things like this).  

Thanks @ruchit2801, and sorry for the slow review!
